commerce_coordinator/apps/iap/api/v1/views.py

```diff
@@ -0,0 +1,188 @@
+"""
+Views for creating orders via the IAP API.
+
+This module handles the order creation process from an existing commercetools cart,
+associates a payment and returns the final order details.
+"""
+
+import logging
+
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+from django.http import JsonResponse
+
+from commercetools.platform.models.order import (
+    OrderFromCartDraft,
+    OrderUpdate,
+    OrderTransitionLineItemStateAction,
+)
+from commercetools.platform.models.payment import (
+    PaymentDraft,
+    PaymentMethodInfo,
+    PaymentStatusDraft,
+)
+from commercetools.platform.models.cart import (
+    CartAddPaymentAction,
+)
+from commercetools.platform.models.common import Reference, ReferenceTypeId
+from commercetools.platform.models.state import StateResourceIdentifier
+from commercetools.exceptions import CommercetoolsError
+
+from commerce_coordinator.apps.commercetools.clients import CommercetoolsAPIClient
+from commerce_coordinator.apps.iap.api.v1.utils import is_cart_active, set_shipping_address
+from commerce_coordinator.apps.iap.api.v1.serializer import CreateOrderSerializer, OrderResponseSerializer
+
+logger = logging.getLogger(__name__)
+
+PAYPAL_PAYMENT_SERVICE_PROVIDER = "PayPal"
+
+class CreateOrderView(APIView):
+    """
+    API view for creating an order from a commercetools cart and payment.
+    Handles validation, optional shipping, PayPal payment creation,
+    payment attachment, and order creation with state transitions.
+    """
+    permission_classes = (IsAuthenticated,)
+
+    # pylint: disable=too-many-statements
+    def post(self, request):
+        """Handles POST request to create an order from a cart and payment."""
+        serializer = CreateOrderSerializer(data=request.data)
+        if not serializer.is_valid():
+            logger.warning(f"Order creation failed due to invalid input: {serializer.errors}")
+            return JsonResponse({'error': serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+
+        cart_id = data["cart_id"]
+        order_number = data["order_number"]
+        payment_method = data["payment_method"]
+        shipping_address_data = data.get("shipping_address")
+        paypal_order = data.get("paypal_order") or {
+            "id": "TEST12345",
+            "status": "APPROVED",
+            "paymentSource": {"type": payment_method or "paypal"}
+        }
+
+        try:
+            ct_client = CommercetoolsAPIClient()
+
+            cart = ct_client.base_client.carts.get_by_id(cart_id)
+
+            if not cart.customer_id or str(cart.customer_id) != str(request.user.customer_id):
```

Comment:
does request.user has 1customer_id`?
i think we should compare using lms_user_id instead